### PR TITLE
ajd Chapter 20

### PR DESCRIPTION
**Pull request by alexdeas**
ajd chapter 20

### CODE REVIEW

File: ajd/My-List-Util/lib/Test/My/List/Util.pm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+=cut
+
+my $Test = Test::Builder->new();
+
+sub sum_ok {
+
+  my ( $actual, $expected, $message ) = @_;
+
+  if ( $actual == $expected ) {
```
Comment (utopia-2) [inline ajd/My-List-Util/lib/Test/My/List/Util.pm:27]: So many warnings.

File: ajd/My-List-Util/t/hosts.t [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  if ( $^O ne 'linux' ) {
+    skip ( 'This is not linux', 1 );
+  }
+
+  file_exists_ok( '/etc/hosts/' );
+
+}
+
+SKIP: {
+  if ( $^O ne 'windows' ) {
```
Comment (utopia-2) [inline ajd/My-List-Util/t/hosts.t:22]: Apparently it never will be, Windows is down as `MSWin32` (see [perlvar#$^O](http://perldoc.perl.org/perlvar.html#%24OSNAME)).

File: ajd/My-List-Util/t/hosts.t [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+plan tests => 2;
+
+diag( "Testing your hosts file, Perl $], $^X" );
+
+SKIP: {
+  if ( $^O ne 'linux' ) {
+    skip ( 'This is not linux', 1 );
+  }
+
+  file_exists_ok( '/etc/hosts/' );
```
Comment (utopia-2) [inline ajd/My-List-Util/t/hosts.t:17]: The exercise wanted us to check for file readability too.

File: ajd/My-List-Util/t/hosts.t [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+use warnings;
+use Test::More;
+use Test::File;
+
+plan tests => 2;
+
+diag( "Testing your hosts file, Perl $], $^X" );
+
+SKIP: {
+  if ( $^O ne 'linux' ) {
```
Comment (utopia-2) [inline ajd/My-List-Util/t/hosts.t:13]: [From @Spinodal's](https://github.com/ardavey/Intermediate-Perl/pull/137/files#r40887241)

File: ajd/My-List-Util/t/hosts.t (hunk 3 above)
Comment (alexdeas) [inline ajd/My-List-Util/t/hosts.t:17]: I'll check you for readability.

Comment (utopia-2) [inline ajd/My-List-Util/t/hosts.t:17]: Sounds like fighting words :fist: :rage: :fist:.

**Pull request merged**